### ISSUE TITLE
Add OpenEdition OPDS catalog to default selection

### ISSUE THREAD

**Issue by digitalethics** [johnfactotum/foliate#448]
OpenEdition offers an [OPDS service](https://www.openedition.org/8881). It would be great to make this [OPDS catalog](http://opds.openedition.org/) available by default to improve ebook discovery for all.

**Comment by digitalethics** [comment]
Linking to #253 for @bertob

**Comment by danielweck** [comment]
For what it's worth, here is a list of "test" and production-grade OPDS feeds:
https://github.com/readium/r2-streamer-js/blob/develop/docs/opds.md#a-selection-of-public-opds-feeds

**Comment by digitalethics** [comment]
> For what it's worth, here is a list of "test" and production-grade OPDS feeds:
> https://github.com/readium/r2-streamer-js/blob/develop/docs/opds.md#a-selection-of-public-opds-feeds

That's a fantastic selection! I think it would be great to maintain a joint database between the Foliate and Thorium reader communities.

**Comment by danielweck** [comment]
Some of the feeds listed there are a bit ... mhmm, how can I say ... "dodgy"? :)
But great for testing edge-cases, because there are numerous border problems like invalid HTTP content types, incorrect HTML/text information, broken links, and other HTTP error status codes, etc.

**Comment by johnfactotum** [comment]
Yeah, the number of high quality OPDS implementations isn't really that high.

It might be a good idea to host the list of catalogs in a separate repo. That way one can update the list of curated catalogs without making a new release. Then,
- it wouldn't add any catalogs by default; instead it would just show the empty placeholder
- when adding a catalog, there would be an option to browse and select from the list of curated catalogs

**Comment by digitalethics** [comment]
> It might be a good idea to host the list of catalogs in a separate repo. That way one can update the list of curated catalogs without making a new release.

This is definitely more efficient. 



> * it wouldn't add any catalogs by default; instead it would just show the empty placeholder
> 
> * when adding a catalog, there would be an option to browse and select from the list of curated catalogs

The current implementation with a selection of default catalogues makes ebook discovery quite easy as you already see them before clicking anywhere. Would it be possible to display a list of curated catalogues from an independent repo in a similar way?

**Comment by johnfactotum** [comment]
> The current implementation with a selection of default catalogues makes ebook discovery quite easy as you already see them before clicking anywhere.

Yeah, but I think there should be some kind of initial "intro to OPDS" welcome message, with a button to add curated catalogs.

- For users new to OPDS, this better informs the user what catalogs are and how they can be used, while still making it easy to explore curated catalogs
- For experienced OPDS users who already have their own list and don't want the default list, they can more easily get to their preferred catalogs without dealing with the default ones

In any case, the curated list should be made available in the "add catalog" dialog or a separate dialog. Currently once you've edited the catalogs in any way, there's no way to recover the default catalogs unless you delete `catalog.json` in the data dir.

**Comment by digitalethics** [comment]
>     * For experienced OPDS users who already have their own list and don't want the default list, they can more easily get to their preferred catalogs without dealing with the default ones

Ideally, there should also be a link within Foliate to encourage anyone who already has OPDS lists to contribute to the public OPDS repo (the idea is to make this "future" repo more discoverable from within Foliate) as this would increase ebook discovery for all. 

I know that not all Foliate users will have a GitHub account. How difficult would it be to implement some kind of **OPDS submission mechanism** from within Foliate? A user clicks "Suggest OPDS source to fellow Foliate users" or something similar, and a bot would then automatically create an issue on this repo without Foliate users having the need to know about this repo and having to sign up for it on GitHub. More technically oriented users can of course contribute via pull requests or raise issues. 

To increase the testing surface for Foliate, the same could be implemented for reporting Foliate issues from within Foliate. A lot of issues, especially from non-technical users, will go unnoticed as many users won't bother or don't have the capability to work with GitHub. In fact, they may never have heard of GitHub. The way they use Foliate, however, matters and their opinions should be heard.
